pontoon/base/views.py

```diff
@@ -1017,6 +1024,35 @@ def download(request):
 @login_required(redirect_field_name='', login_url='/403')
 @require_POST
 @transaction.atomic
+def upload(request):
+    """Upload translated resource."""
+    try:
+        slug = request.POST['slug']
+        code = request.POST['code']
+        part = request.POST['part']
+    except MultiValueDictKeyError:
+        raise Http404
+
+    locale = get_object_or_404(Locale, code=code)
+    if not request.user.has_perm('base.can_translate_locale', locale):
+        return HttpResponseForbidden("Forbidden: You don't have permission to upload files")
+
+    form = forms.UploadFileForm(request.POST, request.FILES)
+
+    if form.is_valid():
+        f = request.FILES['uploadfile']
+        utils.handle_upload_content(slug, code, part, f, request.user)
+        messages.success(request, 'Translations updated from uploaded file.')
+    else:
+        text = form.errors.as_data().itervalues().next()[0][0]
```

Comment:
Also, as a side note, `iteritems` was removed in Python 3 since `items` in Python 3 does the same or better, and in most cases we don't need the savings from using an iterable. I tend to use `items` to make eventual Python 3 porting easier.